components/Graph/index.js

```diff
@@ -67,20 +68,70 @@ const Graph = ({
   openTooltip,
   units,
 }) => {
-  if (typeof document === 'undefined') return null;
+  const firstRun = useRef(true)
+  const shouldFilterBySeek = useRef(false)
+  const [seekDate, setSeekDate] = useState(graph.domain[0])
+  const [data, setData] = useState(graph.data)
+  const [overlayData, setOverlayData] = useState(overlayGraph ? overlayGraph.data : [])
+
+  // run and cut data down if the minimum date range changes
+  useEffect(() => {
+    // do not filter any data on the first run
+    if (firstRun.current) {
+      return
+    }
```

Comment:
seems like a smell to me, but don't know why. is there a way we can set the initial filter to achieve the same thing?

Reply:
The initial filter has the data cut down for each graph from the main graphs component.
This prevents us from cutting that same data again when each graph component mounts, thus not running those operations unnecessarily -- the hooks start listening for seeker filter changes after the graph mounts and execute only if the seeker value or date filter changes, not on initial render.

useRef is used to not trigger any re-renders and avoids re-calculations, this also allows the hook to not have the ref as part of the dependency array and cause possible infinite re-renders or additional re-renders due to an unnecessary state update, and conforms to React's exhaustive dependency rule.